### ISSUE TITLE
White line through the address bar area with 89.0

### ISSUE THREAD

**Issue by conner-s** [mozilla/FirefoxColor#977]
![firefox_uMXzElDq36](https://user-images.githubusercontent.com/32659799/120378905-0a40f900-c2ed-11eb-8609-65c98d00f7ad.png)
![firefox_FdVHZWYrOE](https://user-images.githubusercontent.com/32659799/120378931-1036da00-c2ed-11eb-855b-b2d151df8922.png)

**Comment by Rob--W** [comment]
Could you share the theme that you're using? This may be an issue with Firefox rather than Firefox Color.

**Comment by conner-s** [comment]
Here's my theme. [Link](https://color.firefox.com/?theme=XQAAAAJ-AQAAAAAAAABBqYhm849SCia6aSqEGccwS-xMDPrv2Sw6Caq-qy5QgqeHG4K1461Dh31i_QMEMmIThYYpEv3tk7rtoHTiWSD7k6Mhb64fDGJdLpJ_CysNEor3J5RB3ztY8GHqp_SdSBBzUzZokaRKUusKdlvAAWl1Z4doeLa1jd6K27AsMRMPSk8U8UDun0yz4RMiOZBSGsiNLMF_8qbr8gh9HE3RWO58HCYtqJZYz3NEIs2BMovPY8RTFU7SV3RIyFWab7Zm22q-veXvPnFD6GR7lXkscibWRAW8a_4YWkSebhs___flZHA)

It also happens with any other Firefox Color theme like the preset ones.
![image](https://user-images.githubusercontent.com/32659799/120512803-a88a9880-c399-11eb-918f-2d9fe614d0eb.png)


Though the default Firefox dark theme seems to work fine.
![image](https://user-images.githubusercontent.com/32659799/120513093-e7b8e980-c399-11eb-8172-25e2955d7247.png)

If it's possibly relevant as well I'm on Windows and all my monitors are set to 1440p.

**Comment by Rob--W** [comment]
Tracked upstream at https://bugzilla.mozilla.org/show_bug.cgi?id=1702548

I think that this may be a rounding error that prevented the thin line from being painted in the right color.

**Comment by Nojuuu** [comment]
This happens when your Firefox Color theme has:
Advanced colors --> Toolbar Top Separator --> use Firefox's default style

**Comment by conner-s** [comment]
> This happens when your Firefox Color theme has:
> Advanced colors --> Toolbar Top Separator --> use Firefox's default style

This fixed it, setting the Toolbar Top Separator to Black changed the color of the white bar to black.

**Comment by Nojuuu** [comment]
I think this shouldn't be set to anything by default at all or at least it should be transparent by default or something, because this has really ruined a lot of the old themes on AMO....

And what makes it even worse is that the preview images don't even show if the theme has that separator set.

**Comment by Rob--W** [comment]
> I think this shouldn't be set to anything by default at all or at least it should be transparent by default or something, because this has really ruined a lot of the old themes on AMO....

Themes on AMO don't use Firefox Color. If you see a widespread problem, could you create a new bug report at bugzilla.mozilla.org and link to several themes that have this problem, including screenshots to highlight the issue?

> And what makes it even worse is that the preview images don't even show if the theme has that separator set.

I can consider adding an option if the line is intentional (opposed to a hairline from a rounding error), but otherwise a fix in Firefox would be better.

**Comment by Nojuuu** [comment]
> Themes on AMO don't use Firefox Color. 

Yes, but many themes are usually exported from Firefox Color and then uploaded to AMO. And I think the problem applies to all such themes that had no value set for that separator before Proton update.

In addition, in all themes where this problem exists, the color code for that separator is always `#A0A0A0` / `rgb(160, 160, 160)` , where does it pull that from?

And I actually just went to test and exported Firefox Color theme where that is set to "use Firefox's default style" for that separator and in that xpi file it actually doesn't seem to assign any value to it at all so maybe that is what causes the problem?


> I can consider adding an option if the line is intentional (opposed to a hairline from a rounding error), but otherwise a fix in Firefox would be better.

I meant that the separator should show up in these previews on AMO, because now there's no way to tell if the theme has it before you add it to Firefox.

And just seeing that separator in the preview image could be the deciding factor for someone.

![image](https://user-images.githubusercontent.com/84513173/123427650-9208e300-d5cd-11eb-9948-c623c9a65c44.png)

**Comment by Nojuuu** [comment]
I think that this theme that is currently in the "Recommended" is a good example, because it still shows the old preview image showing that there should be no separators, but if you add it to Firefox, now there are.

![image](https://user-images.githubusercontent.com/84513173/123434285-f9766100-d5d4-11eb-87dc-c62071aeffa3.png)


https://addons.mozilla.org/en-US/firefox/addon/african-heat/
